### PR TITLE
pipeline: move gstreamer pipelines into worker processes

### PR DESCRIPTION
**Pull request by raghaven447**

### CODE REVIEW

File: gaeguli/pipeline.c [hunk 1]
```diff
@@ -92,10 +93,11 @@ typedef enum
   PROP_STREAM_ADAPTOR,
   PROP_GST_PIPELINE,
   PROP_PREFER_HW_DECODING,
+#ifndef USE_PIPELINE_WORKER_PROCESS
```
Comment (xhaakon) [inline gaeguli/pipeline.c]: Why remove these?

File: gaeguli/pipeline.c [hunk 2]
```diff
@@ -795,6 +812,46 @@ _cb_child_watch (GPid pid, gint status, gpointer * data)
...
+    args[i++] = g_strdup_printf ("%d", self->resolution);
+    args[i++] = g_strdup_printf ("%d", self->fps);
+
+    args[i++] = NULL;
+  }
+  return args;
+}
+
+static void
+_gaeguli_free_pipeline_worker_args (gchar ** args)
```
Comment (xhaakon) [inline gaeguli/pipeline.c]: I think you can just use `g_strfreev()`.

File: gaeguli/target.c [hunk 3]
```diff
@@ -678,6 +679,50 @@ _cb_child_watch (GPid pid, gint status, gpointer * data)
   g_spawn_close_pid (pid);
 }
 
+static void
+_gaeguli_free_target_worker_args (gchar ** args)
```
Comment (xhaakon) [inline gaeguli/target.c]: You can use `g_strfreev()` as long as the last item in the list is NULL.

File: gaeguli/workers/gaeguli-pipeline-worker.c [hunk 4]
```diff
@@ -14,18 +23,154 @@ intr_handler (gpointer user_data)
...
+  g_autoptr (GstPluginFeature) feature = NULL;
+  g_autofree gchar *pipeline_str = NULL;
+  GaeguliVideoSource source = GAEGULI_VIDEO_SOURCE_UNKNOWN;
+  gboolean show_overlay = FALSE;
+  gboolean prefer_hw_decoding = FALSE;
+  GaeguliVideoResolution resolution = GAEGULI_VIDEO_RESOLUTION_UNKNOWN;
+  guint fps = 0;
+  gint ret = 0;
+
+  pipeline_str = g_strdup (argv[1]);
```
Comment (xhaakon) [inline gaeguli/workers/gaeguli-pipeline-worker.c]: I'd prefer not passing `pipeline_str` as a cmdline argument. Pass only the values necessary to build it and let the child call `_get_enc_string()`.

File: meson.build [hunk 5]
```diff
@@ -56,6 +56,8 @@ cdata.set_quoted('GAEGULI_PIPELINE_WORKER',
   join_paths(gaeguli_bindir, 'gaeguli_pipeline_worker'))
 cdata.set_quoted('GAEGULI_TARGET_WORKER',
   join_paths(gaeguli_bindir, 'gaeguli_target_worker'))
+cdata.set('GAEGULI_PIPELINE_WORKER_ARGS_NUM', 8)
+cdata.set('GAEGULI_TARGET_WORKER_ARGS_NUM', 11)
```
Comment (xhaakon) [inline meson.build]: Why put these two into `meson.build`? Can't they be defined in `gaeguli-internal.h` for instance?

**Review by xhaakon — CHANGES_REQUESTED** [review]

**Comment by raghaven447** [comment]
Closing this PR as per the revised changes

**Pull request closed without merging**